gpd/gpd-pocket/default.nix

```diff
@@ -0,0 +1,87 @@
+{ lib, pkgs, ... }:
+
+# GPD Pocket support
+#
+# Most of the configuration comes from the Arch Linux Wiki:
+# https://wiki.archlinux.org/index.php/GPD_Pocket
+#
+# Missing:
+# - media keys
+# - brightness control
+# - video over USB-C
+
+{
+  imports = [
+    ../../common/cpu/intel
+    ../../common/pc/laptop
+  ];
+
+  # kernel
+  # TODO: use https://github.com/jwrdegoede/linux-sunxi?
+  boot = {
+    kernelParams = ["fbcon=rotate:1"];
+    kernelPackages = pkgs.linuxPackages_4_14;
+  };
+
+  # wifi
+  nixpkgs.config.allowUnfree = true;
+  hardware.firmware = with pkgs; [
+    brcmfmac4356-firmware
+  ];
+
+  # power management
+  services.tlp = {
+    enable = true;
+    extraConfig = ''
+      DISK_DEVICES="mmcblk0"
+      DISK_IOSCHED="deadline"
+    '';
+  };
+
+  hardware.pulseaudio = {
+    enable = true;
+    extraConfig = ''
+      set-card-profile alsa_card.platform-cht-bsw-rt5645 HiFi
```

Comment:
Is this required to make audio work?

Reply:
There are some hacks required with current mainline kernels. This will change in the future (4.15+).
I am not sure if this is exactly the one since I've been running the 4.15-rc kernels on mine.